### DISCUSSION TITLE
i have issues in download pdf, jpg file

### DISCUSSION THREAD

**Discussion by AnifaDSP** [cypress-io/cypress#27082] [Questions and Help]
While i clicking download button the file getting downloaded, but the cypress will wait for new page load in my application there is no new page for download, how i resolve the issues 
![Screenshot (61)](https://github.com/cypress-io/cypress/assets/121932201/d24cfb4c-b377-4f60-a391-9454981654d5)

**Comment by portyu9** [comment]
Hi @AnifaDSP 

Cypress does not require a new page to download a PDF/JPG. Current Cypress automatically saves browser downloads into `downloadsFolder`, which defaults to `cypress/downloads`.

For example:

```js
cy.contains('button', 'Download').click()

cy.readFile('cypress/downloads/report.pdf', null)
  .should((file) => {
    expect(file.length).to.be.greaterThan(0)
  })
```

Passing `null` to `cy.readFile()` reads a binary file as a `Cypress.Buffer`, which is appropriate for PDFs and images.

If the `.click()` itself is timing out with Cypress saying it is waiting for a new page to load, that indicates the click is probably triggering a **navigation** rather than only a browser download — for example an `<a>`/`window.location` pointing at the file endpoint.

In that case, inspect how the download is implemented. For server downloads, the response should normally be returned as a download (for example with an appropriate `Content-Disposition: attachment` response). Another reliable test is to assert the download URL and use `cy.request()` against that endpoint rather than making the browser navigate to it.

Increasing `pageLoadTimeout` would only hide the underlying navigation issue.

Please mark this answer as accepted if it helped, thank you.
